src/query/expression/src/block.rs

```diff
@@ -590,3 +582,26 @@ impl Clone for Box<dyn BlockMetaInfo> {
         self.clone_self()
     }
 }
+
+fn check_type(data_type: &DataType, value: &Value<AnyType>) {
+    match value {
+        Value::Scalar(Scalar::Null) => {
+            assert!(data_type.is_nullable_or_null());
+        }
+        Value::Scalar(s) => {
+            match s {
+                Scalar::Tuple(ss) => {
```

Comment:
other nested types, such as `Array` and `Map` may have the same problem

Reply:
`Array` and `Map` nest `Column`, maybe don't have the problem.

Reply:
ok, `Array` and `Map` are ok
```sql
mysql> SELECT [true, tt1.b] FROM tt1 CROSS JOIN tt2;
+---------------+
| [true, tt1.b] |
+---------------+
| [1,1]         |
| [1,1]         |
| [1,1]         |
| [1,NULL]      |
| [1,NULL]      |
| [1,NULL]      |
| [1,0]         |
| [1,0]         |
| [1,0]         |
+---------------+
9 rows in set (0.04 sec)
```